src/main/java/de/zalando/ep/zalenium/proxy/DockerSeleniumRemoteProxy.java

```diff
@@ -451,30 +452,54 @@ public synchronized boolean isTestSessionLimitReached() {
         return getAmountOfExecutedTests() >= maxTestSessions;
     }
 
+    public boolean shutdownIfLimitReached() {
+        String currentName = configureThreadName();
+        boolean condition = isTestSessionLimitReached() && !isBusy();
+        if (condition) {
+            String msg = String.format("Exceded more than %s test sessions "
+                    + "in node %s", maxTestSessions, this.getContainerId());
+            LOGGER.debug(msg);
+            timeout(msg, ShutdownType.MAX_TEST_SESSIONS_REACHED);
+        }       
+        setThreadName(currentName);
+        return condition;
+    }
+
     public boolean shutdownIfIdle() {
         String currentName = configureThreadName();
-        boolean testIdle = isTestIdle();
-        boolean testSessionLimitReached = isTestSessionLimitReached();
-        boolean isShutdownIfIdle = testIdle || (testSessionLimitReached && !isBusy());
-        if (isShutdownIfIdle) {
-            LOGGER.debug("Proxy is idle.");
-            timeout("proxy being idle after test.", (testSessionLimitReached ?
-                    ShutdownType.MAX_TEST_SESSIONS_REACHED : ShutdownType.IDLE));
+        boolean condition = isTestIdleMoreThan(900) && !isCleaningUp() && !isBusy();
```

Comment:
Could you please elaborate on the logic here? Why 900 seconds? Are we not checking if the test session limit is reached anymore?

Reply:
This specific timeout of 900 seconds it was to differentiate also another scenario.  For me it's not the same the case where an active session stop to receive commands to the case where the session is closed. We have a penalty of pod creation of about 180 seconds, so